pyuvdata/uvdata.py

```diff
@@ -4628,3 +4654,497 @@ def inflate_by_redundancy(self, tol=1.0, blt_order='time', blt_minor_order=None)
+                                                       atol=self._integration_time.tols[1])))
+
+        if len(inds_to_downsample[0]) == 0:
+            warnings.warn("All values in the integration_time array are already "
+                          "shorter than the value specified; doing nothing.")
+            return
+        # If we're going to do actual work, reorder the baselines to ensure time is
+        # monotonically increasing.
+        # Default of reorder_blts is baseline major, time minor, which is what we want.
+        self.reorder_blts()
+
+        # now re-compute inds_to_downsample, in case things have changed
+        inds_to_downsample = np.nonzero((self.integration_time < min_int_time)
+                                        & (~np.isclose(self.integration_time, min_int_time,
+                                                       rtol=self._integration_time.tols[0],
+                                                       atol=self._integration_time.tols[1])))
+
+        # figure out how many baselines we'll end up with at the end
+        bls_to_downsample = np.unique(self.baseline_array[inds_to_downsample])
+        n_new_samples = 0
+        for bl in bls_to_downsample:
+            bl_inds = np.nonzero(self.baseline_array == bl)[0]
+            n_sample_temp = np.sum(self.integration_time[bl_inds] / min_int_time)
+            if keep_ragged and not np.isclose(n_sample_temp, np.floor(n_sample_temp)):
+                n_new_samples += np.ceil(n_sample_temp).astype(int)
+            else:
+                n_new_samples += np.floor(n_sample_temp).astype(int)
+            # figure out if there are any time gaps in the data
+            # meaning that the time differences are larger than the integration times
+            # time_array is in JD, need to convert to seconds for the diff
+            dtime = np.ediff1d(self.time_array[bl_inds]) * 24 * 3600
+            int_times = self.integration_time[bl_inds]
+            if len(np.unique(int_times)) == 1:
+                # this baseline has all the same integration times
+                if len(np.unique(dtime)) > 1:
+                    warnings.warn("There is a gap in the times of baseline {bl}. "
+                                  "The output may include averages across long "
+                                  "time gaps.".format(bl=self.baseline_to_antnums(bl)))
+                elif not np.isclose(dtime[0], int_times[0]):
+                    warnings.warn("The time difference between integrations is "
+                                  "not the same as the integration time for "
+                                  "baseline {bl}. The output may average across "
+                                  "longer time intervals than "
+                                  "expected".format(bl=self.baseline_to_antnums(bl)))
+
+            else:
+                # varying integration times for this baseline, need to be more careful
```

Comment:
I _think_ one could simply run this else block instead of having the if statement at all.

Reply:
I think you're right, but I've kept it as is because I'm lazy ;-)